src/channel.ts

```diff
@@ -1496,9 +1497,14 @@ export class Channel<
           s.watchers = s.watchers.without(event.user.id);
         }
         break;
+      case 'message.deleted':
+        if (event.message) {
+          if (event.hard_delete) s.removeMessage(event.message as { id: string });
+          else s.addMessageSorted(event.message);
```

Comment:
i wonder if we should update the `removeMessage` type parameter to match `addMessageSorted` so we don't have to cast, unless there's a reason it should stay as it is

Reply:
actually, `removeMessage` param type is more correct than `addMessageSorted`, we should make the  `id` required in `MessageBase` type

Reply:
it definitely was required at one point, not sure why it turned optional, @nhannah do you remember?

Reply:
swapped it to required

Reply:
great! :vir_stache:, let's pretend that worked...